releasenotes/notes/pc-merge-metadata.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+- |
+  **Improved** the `meshConfig.defaultConfig.proxyMetadata` field to do a deep megre when overriden, rather than replacing all values.
```

Comment:
```suggestion
  **Improved** the `meshConfig.defaultConfig.proxyMetadata` field to do a deep merge when overriden, rather than replacing all values.
```